_posts/2023-10-24-filesystem-integrity-elytron-tool.adoc

```diff
@@ -0,0 +1,176 @@
+
+Clone the elytron-examples repo locally:
+```
+git clone https://github.com/wildfly-security-incubator/elytron-examples
+cd elytron-examples/integrity-filesystem-realm
+```
+
+== Server Configuration
+In the following section, we will configure the server to first create a filesystem realm without integrity checking and then use the Elytron tool to add integrity checking to it. We will first connect to the server using the following command: 
+```
+$ WILDFLY_HOME/bin/jboss-cli.sh --connect
+```
+Let's first create a filesystem realm called `fsRealm` in the server configuration directory. We will add an identity, called quickstartUser to the realm, set a password for the identity and add roles for the identity as well. To do that, we will use the following commands: 
+```
+/subsystem=elytron/filesystem-realm=fsRealm:add(path=fs-realm, relative-to=jboss.server.config.dir)
+/subsystem=elytron/filesystem-realm=fsRealm:add-identity(identity=quickstartUser)
+/subsystem=elytron/filesystem-realm=fsRealm:set-password(identity=quickstartUser, digest={password=password123!, realm=fsRealm, algorithm=digest-md5})
+/subsystem=elytron/filesystem-realm=fsRealm:add-identity-attribute(identity=quickstartUser, name=Roles, value=["Admin", "Guest"])
+``` 
+Now if we navigate to the server configuration directory, we will see a new directory created, called `fs-realm`, which contains files relevant to the filesystem realm we just created. If you navigate to the xml file inside the folder, you will see that the details related to the identity we added are all human readable. Additionally, the name of the principal appears in the name of the xml file. An example of the contents of the xml file for a filesystem without integrity checking can be seen below: 
+
+```
+<principal name="quickstartUser"/>
+    <credentials>
+        <password algorithm="digest-md5" format="base64">Ag5xdWlja3N0YXJ0VXNlcgdmc1JlYWxt4Y4F7KmGKSCPrtzjzHD8Hg==</password>
+    </credentials>
+    <attributes>
+        <attribute name="Roles" value="Admin"/>
+        <attribute name="Roles" value="Guest"/>
+    </attributes>
+```
+
+We then create a new security domain which will make use of our filesystem realm as follows:
+```
+/subsystem=elytron/security-domain=fsDomain:add(realms=[{realm=fsRealm}], default-realm=fsRealm,permission-mapper=default-permission-mapper)
+```
+And lastly, update our security domain mapping in the Undertow subsystem:
+```
+/subsystem=undertow/application-security-domain=other:write-attribute(name=security-domain, value=fsDomain)
+```
+Now we can deploy and access the application and log in using the credentials for quickstartUser. However, without any integrity checking, anyone with access to the contents of the filesystem realm can tamper with it. So, let's add integrity checking for this filesystem realm. 
+
+== Adding Integrity Checking
+In order to access the elytron tool, we can use the following command: 
```

Comment:
To create a new filesystem realm based off our existing one but with the integrity support enabled, we can use the following Elytron Tool command:

Reply:
fixed!